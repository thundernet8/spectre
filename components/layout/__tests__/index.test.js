import React from "react";
// import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Layout from "..";

const { Header, Content, Sider, Footer } = Layout;

describe("Layout", () => {
    it("should render Layout", () => {
        // TODO:
    });

    it("renders correctly", () => {
        const tree = renderer
            .create(
                <Layout>
                    <Header>Header</Header>
                    <Layout>
                        <Content>Main content</Content>
                        <Sider>Right sidebar</Sider>
                    </Layout>
                    <Footer>Footer</Footer>
                </Layout>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
