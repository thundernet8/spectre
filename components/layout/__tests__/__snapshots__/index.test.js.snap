// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<div
  className="spt-layout"
>
  <div
    className="spt-layout-header"
  >
    Header
  </div>
  <div
    className="spt-layout spt-layout-has-sider"
  >
    <div
      className="spt-layout-content"
    >
      Main content
    </div>
    <div
      className="spt-layout-sider"
      style={
        Object {
          "flex": "0 0 200px",
          "width": "200px",
        }
      }
    >
      <div
        className="spt-layout-sider-children"
      >
        Right sidebar
      </div>
    </div>
  </div>
  <div
    className="spt-layout-footer"
  >
    Footer
  </div>
</div>
`;
